shared/metrics/src/main/java/io/pravega/shared/metrics/StatsLoggerImpl.java

```diff
@@ -54,9 +55,9 @@ public Counter createCounter(String statName, String... tags) {
     }
 
     @Override
-    public <T extends Number> Gauge registerGauge(final String statName, Supplier<T> valueSupplier, String... tags) {
+    public Gauge registerGauge(final String statName, final Supplier<Number> valueSupplier, String... tags) {
         try {
-            return new GaugeImpl<>(statName, valueSupplier, tags);
+            return new GaugeImpl<>(statName, Preconditions.checkNotNull(valueSupplier), tags);
```

Comment:
`Preconditions.checkNotNull`'s exception will be swallowed up and you'll always return NULLGAUGE. Is this intentional? I think it would make more sense to throw the NullPtrEx to the caller instead of this.

Reply:
Yes it's by design(to have NULLMETRICS for Noop. 
We may have a separate issue for the improvement though.